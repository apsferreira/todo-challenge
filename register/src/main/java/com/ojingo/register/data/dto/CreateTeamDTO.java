package com.ojingo.register.data.dto;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.Size;

import com.sun.istack.NotNull;

public class CreateTeamDTO {

	@NotNull
	@NotEmpty
	@Size(min=3, max=10)
	public String name;
	
	public String key;
	
	public Long idTeam; 
}
