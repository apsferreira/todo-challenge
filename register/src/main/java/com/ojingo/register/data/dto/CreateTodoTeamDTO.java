package com.ojingo.register.data.dto;

import javax.validation.constraints.NotEmpty;

import com.sun.istack.NotNull;

public class CreateTodoTeamDTO {
	@NotNull
	@NotEmpty
	public String description;
}
