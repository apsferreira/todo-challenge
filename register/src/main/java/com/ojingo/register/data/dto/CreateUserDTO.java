package com.ojingo.register.data.dto;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.Size;

import com.sun.istack.NotNull;

public class CreateUserDTO {

	@NotNull
	@NotEmpty
	@Size(min=3, max=10)
	public String name;
	
	@NotNull
	@NotEmpty
	public Long idTeam;
}
