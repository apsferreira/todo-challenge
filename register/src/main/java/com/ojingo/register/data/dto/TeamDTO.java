package com.ojingo.register.data.dto;

import java.util.List;

public class TeamDTO {
	public Long id;
	
	public String name;
	
	public List<UserDTO> users;
	
	public List<TodoDTO> todos;
}
