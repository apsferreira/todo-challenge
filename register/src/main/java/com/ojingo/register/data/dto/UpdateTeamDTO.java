package com.ojingo.register.data.dto;

public class UpdateTeamDTO {
	public String name;
}
