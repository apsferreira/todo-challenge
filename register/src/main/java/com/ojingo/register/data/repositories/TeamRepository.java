package com.ojingo.register.data.repositories;

import javax.enterprise.context.ApplicationScoped;

import com.ojingo.register.domain.entities.Team;

import io.quarkus.hibernate.orm.panache.PanacheRepository;


@ApplicationScoped
public class TeamRepository implements PanacheRepository<Team> {

	public boolean isNew(Team team) {
		if(this.find("name", team.name).count() > 0) {
			return false;
		}
		
		return true;
	}
	
	public Team create(Team team) {
		if (this.isNew(team)) {
			this.persist(team);
		}
		
		return team;		
	}  
	
	public Team update(Team teamToUpdate, Team team) {
		if(team.name != null) {
			teamToUpdate.name = team.name;
		}
		
		this.persist(teamToUpdate);
		
		return teamToUpdate;
	}
}
