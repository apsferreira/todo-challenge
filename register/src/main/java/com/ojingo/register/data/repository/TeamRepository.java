package com.ojingo.register.data.repository;

import javax.enterprise.context.ApplicationScoped;

import com.ojingo.register.domain.models.Team;

import io.quarkus.hibernate.orm.panache.PanacheRepository;


@ApplicationScoped
public class TeamRepository implements PanacheRepository<Team> {

}
