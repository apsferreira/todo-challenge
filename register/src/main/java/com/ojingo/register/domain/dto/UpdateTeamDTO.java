package com.ojingo.register.domain.dto;

public class UpdateTeamDTO {
	public String name;
}
