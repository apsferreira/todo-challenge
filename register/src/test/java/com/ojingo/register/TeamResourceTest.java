package com.ojingo.register;

import javax.inject.Inject;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.ojingo.register.data.repositories.TeamRepository;
import com.ojingo.register.domain.dto.UpdateTeamDTO;
import com.ojingo.register.util.TokenUtils;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.Header;
import io.restassured.http.ContentType;
import io.restassured.specification.RequestSpecification;
import junit.framework.Assert;


@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(DatabaseLifecycleManager.class)
@SuppressWarnings("deprecation")
public class TeamResourceTest {
	
	@Inject
	private TeamRepository teamRepository;
	
	private String token;
	
	@BeforeEach
	public void generateToken() throws Exception {
		token = TokenUtils.generateTokenString("/JWTRegisterClaims.json", null);
	}
	
	private RequestSpecification given() {
		return RestAssured.given().contentType(ContentType.JSON)
				.header(new Header("Authorization", "Bearer " + token));
	}

	@Test
	@DataSet("teams-cenary-1.yml")
	public void testGetAll() {
 		String result = given()
				.when().get("/api/v1/teams")
				.then()
				.statusCode(200)
				.extract().asString();
		Approvals.verifyJson(result);
	}
	
	@Test
	@DataSet("teams-cenary-1.yml")
	public void testGetTeam() {
		Long parameterValue = 123L;
		
 		String result = given()
 				.with().pathParam("idTeam", parameterValue)
				.when().get("/api/v1/teams/{idTeam}")
				.then()
				.statusCode(200)
				.extract().asString();
 		
		Approvals.verifyJson(result);
	}
	
	@Test
	@DataSet("teams-cenary-2.yml")
	public void testGetUsersFromTeam() {
		Long parameterValue = 123L;
		
 		String result = given()
 				.with().pathParam("idTeam", parameterValue)
				.when().get("/api/v1/teams/{idTeam}/users")
				.then()
				.statusCode(200)
				.extract().asString();
 		
		Approvals.verifyJson(result);
	}
	
	@Test
	@DataSet("teams-cenary-1.yml")
	public void testUpdateTeam() {
		UpdateTeamDTO updateTeamDTO = new UpdateTeamDTO();
		
		updateTeamDTO.name = "new Name";
		Long parameterValue = 123L;
		
		given()
			.with().pathParam("id", parameterValue)
				.body(updateTeamDTO)
				.when().put("/api/v1/teams/{id}")
				.then()
				.statusCode(200)
				.extract().asString();
		
		Assert.assertEquals(updateTeamDTO.name, teamRepository.findById(parameterValue).name);
	}
	
	
}
