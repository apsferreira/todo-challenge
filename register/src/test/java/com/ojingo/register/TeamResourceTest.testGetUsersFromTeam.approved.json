[
  {
    "id": 123,
    "team": {
      "id": 123,
      "name": "test"
    },
    "username": "user1"
  },
  {
    "id": 345,
    "team": {
      "id": 123,
      "name": "test"
    },
    "username": "user2"
  }
]